core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -351,13 +351,17 @@ class ControllerBrokerRequestBatch(controller: KafkaController, stateChangeLogge
     addUpdateMetadataRequestForBrokers(controllerContext.liveOrShuttingDownBrokerIds.toSeq, Set(topicPartition))
   }
 
-  def addStopReplicaRequestForBrokers(brokerIds: Seq[Int], topicPartition: TopicPartition, deletePartition: Boolean,
-                                      callback: (AbstractResponse, Int) => Unit) {
+  def addStopReplicaRequestForBrokers(brokerIds: Seq[Int], topicPartition: TopicPartition, deletePartition: Boolean): Unit = {
     brokerIds.filter(_ >= 0).foreach { brokerId =>
+      def callback(stopReplicaResponse: AbstractResponse): Unit = {
+        if (deletePartition)
+          controller.eventManager.put(controller.TopicDeletionStopReplicaResponseReceived(stopReplicaResponse, brokerId))
```

Comment:
I added a check for whether the topic is undergoing deletion. Let me know if that seems sufficient. It would be nice to avoid the awkward callback through the ReplicaStateMachine if possible.